metadata/src/main/java/org/apache/kafka/image/AclsDelta.java

```diff
@@ -61,7 +61,13 @@ public void replay(AccessControlEntryRecord record) {
     }
 
     public void replay(RemoveAccessControlEntryRecord record) {
-        changes.put(record.id(), Optional.empty());
+        if (image.acls().containsKey(record.id())) {
+            changes.put(record.id(), Optional.empty());
+        } else if (changes.containsKey(record.id())) {
+            changes.remove(record.id());
+        } else {
+            throw new RuntimeException("Failed to find existing ACL with ID " + record.id() + " in either image or changes");
+        }
```

Comment:
This comment is for:
```java
public Map<Uuid, Optional<StandardAcl>> changes() {
    return changes;
}
```
Can we document this method? I am particularly interested on the return type. I get the impression that a value of `Optional.empty` should interpreted as a delete.